backend/src/zimfarm_backend/routes/schedules/logic.py

```diff
@@ -0,0 +1,372 @@
+    params: Annotated[SchedulesGetSchema, Query()],
+    session: OrmSession = Depends(gen_dbsession),
+) -> ListResponse[ScheduleLightSchema]:
+    skip = params.skip if params.skip else 0
+    limit = params.limit if params.limit else 20
+    results = db_get_schedules(
+        session,
+        skip=skip,
+        limit=limit,
+        lang=params.lang,
+        categories=params.category,
+        tags=params.tag,
+        name=params.name,
+    )
+    return ListResponse(
+        meta=calculate_pagination_metadata(
+            nb_records=results.nb_records,
+            skip=skip,
+            limit=limit,
+            page_size=len(results.schedules),
+        ),
+        items=cast(list[ScheduleLightSchema], results.schedules),
+    )
+
+
+@router.post("")
+def create_schedule(
+    schedule: ScheduleCreateSchema,
+    session: OrmSession = Depends(gen_dbsession),
+    current_user: User = Depends(get_current_user),
+) -> ScheduleCreateResponseSchema:
+    """Create a new schedule"""
+    if not check_user_permission(current_user, namespace="schedules", name="create"):
+        raise UnauthorizedError("You are not allowed to create a schedule")
+
+    db_schedule = db_create_schedule(
+        session,
+        name=schedule.name,
+        category=ScheduleCategory(schedule.category),
+        language=schedule.language,
+        config=schedule.config,
+        tags=schedule.tags,
+        enabled=schedule.enabled,
+        notification=schedule.notification,
+        periodicity=schedule.periodicity,
+    )
+    return ScheduleCreateResponseSchema(
+        id=db_schedule.id,
+    )
+
+
+@router.get("/backup")
+def get_schedules_backup(
+    session: OrmSession = Depends(gen_dbsession),
+    current_user: User | None = Depends(get_current_user_or_none),
+    hide_secrets: Annotated[bool | None, Query()] = True,  # noqa: FBT002
+) -> JSONResponse:
+    """Get a list of schedules"""
+    if not (
+        current_user
+        and check_user_permission(current_user, namespace="schedules", name="update")
+    ):
+        exclude_notifications = True
+    else:
+        exclude_notifications = False
+
+    if hide_secrets or exclude_notifications:
+        show_secrets = False
+    else:
+        show_secrets = True
+
+    results = get_all_schedules(session)
+    schedules = cast(list[ScheduleFullSchema], results.schedules)
+    content: list[dict[str, Any]] = []
+    for schedule in schedules:
+        if exclude_notifications:
+            schedule.notification = None
+
+        content.append(
+            schedule.model_dump(mode="json", context={"show_secrets": show_secrets})
+        )
+
+    return JSONResponse(content=content)
+
+
+@router.get("/{schedule_name}")
+def get_schedule(
+    schedule_name: Annotated[ScheduleNameField, Path()],
+    session: OrmSession = Depends(gen_dbsession),
+    current_user: User | None = Depends(get_current_user_or_none),
+    hide_secrets: Annotated[bool | None, Query()] = True,  # noqa: FBT002
```

Comment:
Do we really need this `FBT002` hint? I think that parameters are always passed by name by FastAPI, so we could add the `*, hide_secrets: ...` that `FBT002` want us to add for boolean args (to force passing parameter by name)